march_data_collector/src/march_data_collector/cp_calculator.py

```diff
@@ -1,70 +1,114 @@
+
+        current_time = updated_center_of_mass.header.stamp
+        self._delta_t = (current_time - self._prev_t).to_sec()
+
+        if self._delta_t == 0:
+            return
+
+        self.vx = (updated_center_of_mass.pose.position.x - self._center_of_mass_marker.pose.position.x) / self._delta_t
+        self.vy = (updated_center_of_mass.pose.position.y - self._center_of_mass_marker.pose.position.y) / self._delta_t
+
+        self.x = self._center_of_mass_marker.pose.position.x
+        self.y = self._center_of_mass_marker.pose.position.y
+        self.z = self._center_of_mass_marker.pose.position.z
+
+        self._center_of_mass_marker = updated_center_of_mass
+        self._prev_t = updated_center_of_mass.header.stamp
+
+    def _calculate_capture_point(self, duration):
+        """Calculate a future capture point pose using the inverted pendulum and center of mass.
+
+        :param duration:
+            the amount of seconds away from the current time the capture point should be calculated
+        """
+        rospy.logdebug('time diff: {td}'.format(td=str(self._delta_t)))
+        if self._delta_t == 0:
+            rospy.loginfo('Calculating future center of mass for capture point pose.')
+            new_center_of_mass = InvertedPendulum.numeric_solve_to_t(self.x, self.y, self.z, self.vx, self.vy, duration)
 
             try:
-                trans = self.tf_buffer.lookup_transform('world', self.foot_link, rospy.Time())
-                try:
-                    multiplier = sqrt(com_mark.pose.position.z / self.g)
-                except ValueError:
-                    rospy.logdebug_throttle(1, 'Cannot calculate capture point, because center of mass height is '
-                                               'smaller than 0')
-                    return self.marker
-
-                x_cp = trans.transform.translation.x + x_dot * multiplier
-                y_cp = trans.transform.translation.y + y_dot * multiplier
-
-                self.update_marker(x_cp, y_cp)
-
-                self.prev_x = com_mark.pose.position.x
-                self.prev_y = com_mark.pose.position.y
-                self.prev_t = current_time
+                world_transform = self._tf_buffer.lookup_transform('world', self._foot_link, rospy.Time())
+                if new_center_of_mass.z <= 0:
+                    rospy.logdebug_throttle(1, 'Cannot calculate capture point; center of mass < 0')
+
+                capture_point_multiplier = sqrt(new_center_of_mass.z / self._gravity_constant)
+
+                x_cp = world_transform.transform.translation.x + new_center_of_mass.vx * capture_point_multiplier
+                y_cp = world_transform.transform.translation.y + new_center_of_mass.vy * capture_point_multiplier
+
+                self._capture_point_marker.header.stamp = rospy.get_rostime()
+                self._capture_point_marker.pose.position.x = x_cp
+                self._capture_point_marker.pose.position.y = y_cp
```

Comment:
```suggestion
                self._capture_point_marker.pose.position.x = x_cp + world_transform.transform.translation.x
                self._capture_point_marker.pose.position.y = y_cp + world_transform.transform.translation.y
```
And transform them back to the world frame